src/popover/PopOverContext.tsx

```diff
@@ -0,0 +1,87 @@
+import React, {createContext, useCallback, useContext, useState} from 'react';
+import {PopOverKey} from './types';
+import {PopOver} from '@atb/components/popover';
+import {useOneTimePopover} from './use-one-time-popover';
+import OneTimePopOverTexts from '@atb/translations/components/OneTimePopOver';
+import {useTranslation} from '@atb/translations';
+import {useIsScreenReaderEnabled} from '@atb/utils/use-is-screen-reader-enabled';
+
+export const POPOVER_ANIMATION_DURATION = 200;
+
+type PopOverType = {
+  target: React.RefObject<JSX.Element | null>;
+  oneTimeKey: PopOverKey;
+};
+
+export type PopOverContextType = {
+  addPopOver: (popOver: PopOverType) => void;
+};
+
+const PopOverContext = createContext<PopOverContextType>({
+  addPopOver: () => {},
+});
+
+export const PopOverContextProvider: React.FC = ({children}) => {
+  // Queue of popovers to display
+  const [popOvers, setPopOvers] = useState<PopOverType[]>([]);
+  const {isSeen, setPopOverSeen} = useOneTimePopover();
+  const {t} = useTranslation();
+  const isScreenReaderEnabled = useIsScreenReaderEnabled();
+  const current = popOvers[0];
+
+  const addPopOver = useCallback(
+    (popOver: PopOverType) => {
+      if (!isSeen(popOver.oneTimeKey)) {
+        // Setting a timeout before adding the popover to allow potential page transitions
+        // or similar animations to complete before the popover is displayed.
```

Comment:
Should the timeout be removed or optional?
Ideally `addPopOver` will be called when everything is ready for it to be shown?

Reply:
Solved with `InteractionManager.runAfterInteractions()`